packages/main/src/plugin/extension-loader.ts

```diff
@@ -562,9 +567,12 @@ export class ExtensionLoader {
         this.watcherExtensions.set(extension.id, extensionWatcher);
       }
 
-      const runtime = this.loadRuntime(extension);
-
-      await this.activateExtension(extension, runtime);
+      // activate the extension if it hasn't been disabled
+      if (!this.getDisabledExtensionIds().includes(extension.id)) {
```

Comment:
it doesn't feel natural to store disabled state in a separate map/array

we already map of extensions, ids, etc.

Probably the state should go to the extension object directly. It would avoid another indirection level and keep consistency between extensions and the disabled extensions.

I would avoid usage  of `double negation` like instead of not disabled ((!this.getDisabledExtensionIds), check enabled and if state of enablement is in the extension it's easier.

like `if (extension.isEnabled() {
   await this.activateExtension()
`